api/v1alpha7/openstackmachine_types.go

```diff
@@ -107,6 +107,11 @@ type OpenStackMachineStatus struct {
 	// +optional
 	InstanceState *InstanceState `json:"instanceState,omitempty"`
 
+	// SecurityGroups is the list of security groups that were applied to the ports
```

Comment:
Note for myself: in comment I'll explain that we only populate that once all the rules have been applied.

Note for reviewers: do we want to have another field `SecurityGroupsReady` once the reconciler has finished its job successfully? Now, let's say we managed to successfully reconcile once and `Status.SecurityGroups` was populated once, then somehow the reconciler fails but we still have `Status.SecurityGroups`, we have [mark](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/pull/1774/files#diff-e0a0984c0e26655878e2f94623fbf8cddb4d8a89e3443c1296dfaadc8802838aR339) the instance as NotReady. Is that good enough?

Reply:
> For machine creation I think it's clear that we would only set ready after security groups have been applied.

Exactly. That's what I meant when writing that code:

```golang
err = r.reconcileSecurityGroupsToInstance(scope.Logger(), openStackCluster, machine, openStackMachine, instanceStatus, computeService, networkingService)
if err != nil {
	conditions.MarkFalse(openStackMachine, infrav1.InstanceReadyCondition, infrav1.SecurityGroupApplyFailedReason, clusterv1.ConditionSeverityError, "Applying security groups failed: %v", err)
	return ctrl.Result{}, fmt.Errorf("add security groups to instance: %w", err)
}
```

> For an update to applied security groups it's less clear, and the CAPI definition doesn't indicate if setting it to false after having set it to true is meaningful or allowed. I would not be inclined to set it to false during a rollout of new managed security groups. I think this would be better indicated in the cluster status as a summary of how many machines have been updated.

ok I'll try to think how the state of how many machines have been updated can be reflected in the `OpenStackClusterStatus`.